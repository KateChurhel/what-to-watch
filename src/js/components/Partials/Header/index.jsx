// libraries
import React, { useState } from 'react';
// views
import SearchBar from '../SearchBar';
// constants
import CATEGORIES from '../../../constants/categories';

const Header = () => {
  const [isMobileNavOpened, setIsMobileNavOpened] = useState(false);

  const closeMobileNav = () => {
    setIsMobileNavOpened(false);
    document.body.classList.remove('nav-mobile-scroll-lock');
  };

  const clickHamburgerButton = () => {
    if (isMobileNavOpened) {
      closeMobileNav();
    } else {
      setIsMobileNavOpened(true);
      document.body.classList.add('nav-mobile-scroll-lock');
    }
  };

  return (
    <header className={`page-header ${isMobileNavOpened ? 'page-header__show-menu' : ''}`}>
      <div className="container">
        <a className="logo" href="/">WTW</a>
        <div className="page-header__menu-wrapper">
          <nav>
            <ul className="page-nav">
              <li className="page-nav__item">
                <a className="page-nav__link" href={`/${CATEGORIES.movie}/1`}>Movies</a>
              </li>
              <li className="page-nav__item">
                <a className="page-nav__link" href={`/${CATEGORIES.tv}/1`}>TV Shows</a>
              </li>
            </ul>
          </nav>
          <SearchBar closeMobileNav={closeMobileNav} />
        </div>
        <button className="page-nav__hamburger-button" onClick={clickHamburgerButton} type="button">
          <span />
        </button>
      </div>
    </header>
  );
};

export default Header;
