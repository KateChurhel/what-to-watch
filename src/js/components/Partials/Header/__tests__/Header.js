import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../index';

describe('Header: ', () => {
  const tree = shallow(<Header />);

  it('correctly renders', () => {
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('open mobile menu', () => {
    tree.find('.page-nav__hamburger-button').simulate('click');
    expect(toJSON(tree)).toMatchSnapshot();
    expect(tree.find('.page-header.page-header__show-menu').exists()).toEqual(true);
  });

  it('close mobile menu', () => {
    const button = tree.find('.page-nav__hamburger-button').simulate('click');
    button.simulate('click');
    button.simulate('click');
    expect(tree.find('.page-header.page-header__show-menu').exists()).toEqual(false);
  });
});
