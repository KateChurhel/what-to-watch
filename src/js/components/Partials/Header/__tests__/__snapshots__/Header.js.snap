// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header:  correctly renders 1`] = `
<header
  className="page-header "
>
  <div
    className="container"
  >
    <a
      className="logo"
      href="/"
    >
      WTW
    </a>
    <div
      className="page-header__menu-wrapper"
    >
      <nav>
        <ul
          className="page-nav"
        >
          <li
            className="page-nav__item"
          >
            <a
              className="page-nav__link"
              href="/movie/1"
            >
              Movies
            </a>
          </li>
          <li
            className="page-nav__item"
          >
            <a
              className="page-nav__link"
              href="/tv/1"
            >
              TV Shows
            </a>
          </li>
        </ul>
      </nav>
      <SearchBar
        closeMobileNav={[Function]}
      />
    </div>
    <button
      className="page-nav__hamburger-button"
      onClick={[Function]}
      type="button"
    >
      <span />
    </button>
  </div>
</header>
`;

exports[`Header:  open mobile menu 1`] = `
<header
  className="page-header page-header__show-menu"
>
  <div
    className="container"
  >
    <a
      className="logo"
      href="/"
    >
      WTW
    </a>
    <div
      className="page-header__menu-wrapper"
    >
      <nav>
        <ul
          className="page-nav"
        >
          <li
            className="page-nav__item"
          >
            <a
              className="page-nav__link"
              href="/movie/1"
            >
              Movies
            </a>
          </li>
          <li
            className="page-nav__item"
          >
            <a
              className="page-nav__link"
              href="/tv/1"
            >
              TV Shows
            </a>
          </li>
        </ul>
      </nav>
      <SearchBar
        closeMobileNav={[Function]}
      />
    </div>
    <button
      className="page-nav__hamburger-button"
      onClick={[Function]}
      type="button"
    >
      <span />
    </button>
  </div>
</header>
`;
